sisl/viz/plots/bands.py

```diff
@@ -281,6 +284,14 @@ def _get_frame_names(self):
             return [childPlot.get_setting("bands_file").name for childPlot in self.child_plots]
 
         return cls.animated("bands_file", bands_files, frame_names = _get_frame_names, wdir = wdir, **kwargs)
+    
+    @property
+    def bands(self):
+        return self.bands_data["E"]
+
+    @property
+    def spin_moments(self):
+        return self.bands_data["spin_moments"].sel(spin=0)
```

Comment:
This just seems weird? spin-moments but selecting spin=0?

Reply:
Maybe `self.bands_data.sel(spin=0)["spin_moments"]` is more understandable?

Basically the spin coordinate is always there so that I don't have to treat each case differently, even though in the non-colinear case it doesn't make much sense.

Here is how the dataset looks for a non-colinear fatbands plot:
```
<xarray.Dataset>
Dimensions:       (axis: 3, band: 56, k: 301, orb: 28, spin: 1)
Coordinates:
  * orb           (orb) int32 0 1 2 3 4 5 6 7 8 9 ... 19 20 21 22 23 24 25 26 27
  * band          (band) int32 0 1 2 3 4 5 6 7 8 ... 47 48 49 50 51 52 53 54 55
  * axis          (axis) <U1 'x' 'y' 'z'
  * k             (k) float64 0.0 0.007169 0.01434 0.02151 ... 2.14 2.147 2.155
  * spin          (spin) int64 0
Data variables:
    E             (spin, k, band) float64 -24.83 -24.83 -24.81 ... 19.67 19.67
    spin_moments  (spin, k, band, axis) float64 -0.2438 0.1371 ... 0.9072 0.1044
    weight        (spin, k, band, orb) float64 3.708e-05 0.0001171 ... 0.217
Attributes:
    ticks:       [0.         0.78860842 1.69921498 2.15451827]
    ticklabels:  ['M', '$\\Gamma$', 'K', 'M']
    bz:          BandStructure{nk: 301,\n SuperCell{nsc: [5 5 1],\n  A=[4.600...
    parent:      Hamiltonian{non-zero: 6476, orthogonal: False,\n Spin{spin-o...
```

Reply:
Perhaps spin shouldn't be there for non-diagonal spins?

Reply:
And in non-polarized calculations?

Reply:
yeah, always a problem... ;)
In my PDOS returns I don't return spin-dimension for non-polarized. This is also annoying since users have to change code to handle corner cases. But I don't know how we should do it more easily...

Reply:
Ok, then I'll have to look at it carefully, because this implies a handful of modifications in the code and in tests, basically adding checks to discern between the two cases.

Reply:
Ok, hmm.. I don't really know. If this is not something the user really touches then ok. However, I think we need to make it "easy to understand for users" and having a spin-index for a non-polarized calculation and non-collinear is a bit problematic. I should probably change the spin index in the total, x, y, z returns to spin-components for clarity and reduction of confusion? Any suggestions here?

Reply:
> I should probably change the spin index in the total, x, y, z returns to spin-components for clarity and reduction of confusion? Any suggestions here?

I don't understand where do you mean to change it :sweat_smile:

Reply:
For instance the dataarray returned from sisl.io.siesta.pdos, this contains a spin index, however it is named so it is not ambiguous.

Reply:
All I mean is that there are other places in sisl where I should reconsider what to return and more importantly, _how_.